ztools/ztls/common.go

```diff
@@ -102,12 +104,81 @@ const (
 	CurveP521 CurveID = 25
 )
 
+func (curveID *CurveID) MarshalJSON() ([]byte, error) {
+	buf := make([]byte, 2)
+	buf[0] = byte(*curveID >> 8)
+	buf[1] = byte(*curveID)
+	enc := strings.ToUpper(hex.EncodeToString(buf))
+	aux := struct {
+		Hex   string `json:"hex"`
+		Name  string `json:"name"`
+		Value uint16 `json:"value"`
+	}{
+		Hex:   fmt.Sprintf("0x%s", enc),
+		Name:  curveID.String(),
+		Value: uint16(*curveID),
+	}
+
+	return json.Marshal(aux)
+}
+
+func (curveID *CurveID) UnmarshalJSON(b []byte) error {
+	aux := struct {
+		Hex   string `json:"hex"`
+		Name  string `json:"name"`
+		Value uint16 `json:"value"`
+	}{}
+	if err := json.Unmarshal(b, &aux); err != nil {
+		return err
+	}
+	if expectedName := nameForCurve(aux.Value); expectedName != aux.Name {
+		return fmt.Errorf("mismatched curve and name, curve: %d, name: %s, expected name: %s", aux.Value, aux.Name, expectedName)
+	}
+	*curveID = CurveID(aux.Value)
+	return nil
+}
+
+type PointFormat uint8
+
```

Comment:
Did you make this change? Didn't Zakir implement this a while ago? Is something up with this merge? Are you rebased onto the most recent master?